uvicorn/server.py

```diff
@@ -44,16 +44,19 @@ def __init__(self, config):
         self.last_notified = 0
 
     def run(self, sockets=None):
-        self.config.setup_event_loop()
+        config = self.config
+        config.setup_event_loop()
+
+        if not config.loaded:
+            config.load()
```

Comment:
Let's add a comment about why we're doing this here, perhaps?

```suggestion
        # Setup and load app before entering the async environment, so it's possible to
        # run any async setup on our side or in the app module.
        # See: https://github.com/encode/uvicorn/issues/941
        config = self.config
        config.setup_event_loop()

        if not config.loaded:
            config.load()
```